### PR TITLE
Redo Arm DAC implementation for additive, wavetable synthesis, sample playback

### PR DESCRIPTION
**Pull request by jackhumbert**
## Description

This allows us to take full control of the DAC and output playback 44.1khz (standard audio definition) with two voices (this is adjustable - quality and number of voices have an inverse relationship). There are three different implementations that can be used - I have two of them delegated to examples in keymaps, but they are:

* Additive Synthesis - combining simple (or complex) waves to make sounds
* Wavetable Synthesis - modulating between waves based on time, encoder position, etc
* Sample Playback - playback of a recorded sample without pitch modulation, etc (not implemented but is possible)

The implementation I have here is compatible with https://waveeditonline.com/ and their downloadable editor - I think this is pretty powerful, as people can preview the sound in their editor, and download tables to mess with. The are two scripts to convert .wav files into the appropriate array declarations to be included - I'm not sure if we should be storing such large files in the repo though (the ones on linked site are all public domain).

I need to write a lot of documentation for this, but I wanted to get some thoughts on it before that.

## Types of Changes

- [x] Core
- [x] New feature
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

### CODE REVIEW

File: quantum/audio/audio_arm.c [hunk 1]
```diff
@@ -77,200 +62,198 @@ bool glissando = true;
...
   [DAC_BUFFER_SIZE/2 ... DAC_BUFFER_SIZE  -1] = 0,
 };
 
-// squarewave
-static const dacsample_t dac_buffer_2[DAC_BUFFER_SIZE] = {
-  // opposite of dac_buffer above
-  [0                 ... DAC_BUFFER_SIZE/2-1] = 0,
-  [DAC_BUFFER_SIZE/2 ... DAC_BUFFER_SIZE  -1] = DAC_SAMPLE_MAX,
-};
+static dacsample_t dac_buffer_empty[DAC_BUFFER_SIZE] = { DAC_OFF_VALUE };
```
Comment (drashna) [inline quantum/audio/audio_arm.c]: Looks like this is causing issues, for every board but the Planck EZ.

Comment (drashna) [reply quantum/audio/audio_arm.c]:
```suggestion
#if defined(A4_AUDIO) || defined(A5_AUDIO)
static dacsample_t dac_buffer_empty[DAC_BUFFER_SIZE] = { DAC_OFF_VALUE };
#endif
```

Comment (drashna) [reply quantum/audio/audio_arm.c]: That or, I think we need to define A4 or A5 for audio on each and every board.

Comment (jackhumbert) [reply quantum/audio/audio_arm.c]: Yeah, that's something I was debating - I think it makes sense, since that's how we do the AVR audio.

Comment (drashna) [reply quantum/audio/audio_arm.c]: Probably best, at least for consistency sake then. 

But otherwise, looks good. :)

**Comment by drashna** [comment]
Testing this branch, I'm getting weird issues.  The songs don't play quite right (but they sound a lot closer to what AVR sounds like), and I've had my board lock up a couple of times during the startup sound.   I'm not sure what is doing on there to cause that, but let me know if you need/want me to test anything

**Comment by JohSchneider** [comment]
FYI: pushed some updates to this branch recently, which adds pwm-audio for "lesser arms" (stm32f1xx) and renames the audio_* sources to reflect their platform-dependence

hope you don't mind that i chime in on the "lets work some on the arm audio implementation"

**Comment by tzarc** [comment]
@jackhumbert are we cool to close this, given #6165 is a bit more comprehensive, now?

**Comment by tzarc** [comment]
Letting stalebot do its thing, as it looks like #6165 is the preferred approach, given the extra configurability.

**Comment by tzarc** [comment]
Perhaps this needs to be taken instead of #6165, seeing as it looks like it's stalled again.

**Pull request closed without merging**